### DISCUSSION TITLE
Global accountant: Message signature format doesn’t follow guidelines for guardian key usage

### DISCUSSION THREAD

**Discussion by tbjump** [wormhole-foundation/wormhole#2292] [Protocol / feature design discussion]
## Status Quo

The Guardian key is being used to sign a JSON message about an observation to be submitted to the Accounting contract. 

## Issue

The signature format doesn’t follow the recommendation of [the whitepaper](https://github.com/wormhole-foundation/wormhole/blob/main/whitepapers/0009_guardian_key.md), which is to prepend a message type identifier of `>=34` bytes. 

## Recommendation

In this line of code: 

[[wormhole/accounting.go at 7f512425dcf5425957c6921ed130caa9569023ab · wormhole-foundation/wormhole](https://github.com/wormhole-foundation/wormhole/blob/7f512425dcf5425957c6921ed130caa9569023ab/node/pkg/accounting/accounting.go#L330)](https://github.com/wormhole-foundation/wormhole/blob/7f512425dcf5425957c6921ed130caa9569023ab/node/pkg/accounting/accounting.go#L330)

I propose that instead of computing `digest := Keccak256Hash(Keccak256Hash(observation))`, compute `digest := Keccak256Hash(ACCOUNTING_OBSERVATION_PREFIX + observation)` , where `ACCOUNTING_OBSERVATION_PREFIX = ACCOUNTING_OBSERVATION_00000000000`

The corresponding changes also need to be made in the Accounting contract. 

Finally, this additional guardian key usage should be documented in the Whitepaper [https://github.com/wormhole-foundation/wormhole/blob/main/whitepapers/0009_guardian_key.md](https://github.com/wormhole-foundation/wormhole/blob/main/whitepapers/0009_guardian_key.md)

**Comment by aadam-10** [comment]
implemented in https://github.com/wormhole-foundation/wormhole/pull/2322
